src/neighbors.cc

```diff
@@ -1159,6 +1159,12 @@ peerRefreshDNS(void *data)
         return;
     }
 
+    peerRefreshDNSNow();
+}
+
+static void
+peerRefreshDNSNow()
```

Comment:
Let's honor modern Squid CamelCase naming guidelines and be more specific/precise (the refresh does not really happen "now" in many cases because we wait for DNS responses before calling peerDNSConfigure()):

```suggestion
peerDnsRefreshStart()
```

The other/old peerRefreshDNS() function, if it survives eventDelete()-related fix we have discussed during the meeting, may be called peerDnsRefreshCheck().

You may want to add a peerScheduleDnsRefreshCheck(delayInSeconds) function that will delete the previous peerDnsRefreshCheck event before scheduling another one. That function will also reduce code duplication among eventAddIsh() callers.

Reply:
Done (54181839).

> reduce code duplication among eventAddIsh() callers

We have one more eventAddIsh() in peerDNSConfigure(), scheduling netdbExchangeStart(). This event is not deleted if a duplicate is going to be added.